pkg/apis/hive/v1/hiveconfig_types.go

```diff
@@ -78,6 +78,26 @@ type HiveConfigSpec struct {
 	// DisabledControllers allows selectively disabling Hive controllers by name.
 	// The name of an individual controller matches the name of the controller as seen in the Hive logging output.
 	DisabledControllers []string `json:"disabledControllers,omitempty"`
+
+	// controllerGoroutines is used to configure the number of goroutines for each controller
+	// +optional
+	ControllerGoroutines *ControllerGoroutinesConfig `json:"controllerGoroutines,omitempty"`
```

Comment:
Can we use "ConcurrentReconciles" here instead of "Goroutines"? The former conveys a concept that the user may care about whereas the latter is an implementation detail.